grails-app/controllers/org/bbop/apollo/OrganismController.groovy

```diff
@@ -123,6 +126,169 @@ class OrganismController {
+                        File newFile = new File(organism.directory + File.separator + fileName)
+
+                        System.out.println("file unzip : "+ newFile.getAbsoluteFile())
+
+                        //create all non exists folders
+                        //else you will hit FileNotFoundException for compressed folder
+                        new File(newFile.getParent()).mkdirs()
+
+                        FileOutputStream fos = new FileOutputStream(newFile)
+
+                        int len;
+                        while ((len = zis.read(buffer)) > 0) {
+                            fos.write(buffer, 0, len)
+                        }
+
+                        fos.close()
+                        ze = zis.getNextEntry()
+                    }
+                    zis.closeEntry()
+                    zis.close()
+
+                    System.out.println("Done exporting file  . . . . loading ref seqs")
+                    sequenceService.loadRefSeqs(organism)
+                    System.out.println("Done loading ref seqs")
+                    if (checkOrganism(organism)) {
+                        organism.save(failOnError: true, flush: true, insert: true)
+                    }
+                } catch (e) {
+                    log.error(e)
+                }
+
+                render findAllOrganisms()
+            } else {
+                def error = [error: 'not authorized to add organism']
+                render error as JSON
+                log.error(error.error)
+            }
+        } catch (e) {
+            def error = [error: 'problem saving organism: ' + e]
+            render error as JSON
+            e.printStackTrace()
+            log.error(error.error)
+        }
+    }
+
+    @RestApiMethod(description = "Adds a track to an existing organism returning a JSON array of all organisms", path = "/organism/addTrackToOrganism", verb = RestApiVerb.POST)
+    @RestApiParams(params = [
+            @RestApiParam(name = "username", type = "email", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "commonName", type = "string", paramType = RestApiParamType.QUERY, description = "A common name (or ID) used for the organism")
+            , @RestApiParam(name = "trackData", type = "string", paramType = RestApiParamType.QUERY, description = "Track data (JBrowse JSON)")
+            , @RestApiParam(name = "type", type = "string", paramType = RestApiParamType.QUERY, description = "Track type")
+            , @RestApiParam(name = "label", type = "string", paramType = RestApiParamType.QUERY, description = "Track label")
+            , @RestApiParam(name = "key", type = "string", paramType = RestApiParamType.QUERY, description = "Track key")
+            , @RestApiParam(name = "category", type = "string", paramType = RestApiParamType.QUERY, description = "(optional) Track metadata (JSON)")
```

Comment:
This seems like a lot of parameters, surely it would be as simple as trackData + a similarly posted file? Anyone who is posting to this API is familiar with JBrowse and already have their configuration in that format, requiring them to separate out the parameters seems strange to me. I jad assumed you would just accept the JSON from the API + write directly to a trackList.json.